starky/src/proof.rs

```diff
@@ -56,4 +89,34 @@ impl<F: RichField + Extendable<D>, const D: usize> StarkOpeningSet<F, D> {
             quotient_polys: eval_commitment(zeta, quotient_commitment),
         }
     }
+
+    // Note: Can't implement this directly on `Challenger` as it's in a different crate.
+    pub fn observe<H: Hasher<F>>(&self, challenger: &mut Challenger<F, H>) {
+        let StarkOpeningSet {
+            local_values,
+            next_values,
+            permutation_zs,
+            quotient_polys,
+        } = self;
+        for v in &[local_values, next_values, permutation_zs, quotient_polys] {
+            challenger.observe_extension_elements(v);
+        }
+    }
+
+    pub(crate) fn to_fri_openings(&self) -> FriOpenings<F, D> {
+        let zeta_batch = FriOpeningBatch {
+            values: [
+                self.local_values.as_slice(),
+                self.quotient_polys.as_slice(),
+                self.permutation_zs.as_slice(),
+            ]
+            .concat(),
+        };
+        let zeta_right_batch = FriOpeningBatch {
+            values: self.next_values.to_vec(),
```

Comment:
I think we'll also want to open `permutation_zs` at `g zeta` right?

Reply:
Good point, I added a `permutation_zs_right` field to `StarkOpeningSet` and fixed that.